tests/test_serialization.py

```diff
@@ -136,6 +148,90 @@ def test_pickle_list() -> None:
         fp.seek(0)
         c1 = pickle.load(fp)
         assert c == c1
+        assert get_ref_type(c1) == Optional[List[Any]]
+        assert c1._metadata.element_type is Any
+        assert c1._metadata.optional is True
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "input_,node,element_type,key_type,optional,ref_type",
+    [
+        (GenericList, "list", Any, Any, False, List[Any]),
+        (OptGenericList, "opt_list", Any, Any, True, Optional[List[Any]]),
+        (GenericDict, "dict", Any, Any, False, Dict[Union[str, Enum], Any]),
+        (
+            OptGenericDict,
+            "opt_dict",
+            Any,
+            Any,
+            True,
+            Optional[Dict[Union[str, Enum], Any]],
+        ),
+        (SubscriptedDict, "dict", int, str, False, Dict[str, int]),
+        (SubscriptedList, "list", int, Any, False, List[int]),
+        (
+            DictConfig(content={"a": "foo"}, element_type=str, key_type=str),
+            None,
+            str,
+            str,
+            True,
+            Optional[Dict[str, str]],
+        ),
+        (
+            DictConfig(content={"a": "foo"}, ref_type=Dict[str, str]),
+            None,
+            str,
+            str,
+            True,
+            Optional[Dict[str, str]],
+        ),
+        (
+            ListConfig(content=[1, 2], element_type=int),
+            None,
+            int,
+            Any,
+            True,
+            Optional[List[int]],
+        ),
+        (
+            ListConfig(content=[1, 2], ref_type=List[int]),
+            None,
+            int,
+            Any,
+            True,
+            Optional[List[int]],
+        ),
+    ],
+)
+def test_pickle_generic(
```

Comment:
did you move this function?
can you not do it, it makes it harder to review.

Reply:
Sorry, rebasing I missed that test_pickle_generic was duplicated and I had to remove one of them. It looks like I moved it. I'm fixing it.

Reply:
Still not moved back.